include/gz/utils/Subprocess.hh

```diff
@@ -32,46 +36,80 @@ namespace gz
 namespace utils
 {
 
+/// \brief Create a RAII-type object that encapsulates a subprocess.
 class Subprocess
 {
+  /// \brief Constructor
+  ///
+  /// This variant will spawn a subprocess that inherits the environment
+  /// from the calling process.
+  ///
+  /// \param[in] _commandLine set of arguments starting with an executable
+  ///   used to spawn the subprocess
+  public: explicit Subprocess(const std::vector<std::string> &_commandLine):
+    commandLine(_commandLine),
+    environment({}),
+    inheritEnvironment(true)
+  {
+    this->Create();
+  }
+
+  /// \brief Constructor
+  ///
+  /// This variant will spawn a subprocess that uses the user-specified
+  /// environment
+  ///
+  /// \param[in] _commandLine set of arguments starting with an executable
+  ///   used to spawn the subprocess
+  /// \param[in] _environment environment variables to set in the spawned
+  ///   subprocess
   public: Subprocess(const std::vector<std::string> &_commandLine,
-                     const std::vector<std::string> &_environment = {}):
+                     gz::utils::EnvironmentMap _environment):
```

Comment:
I like this signature, but would it does break API. Would it be possible to add `Subprocess(const std::vector<std::string> &_commandLine,  const std::vector<std::string> &_environment`)` that would turn around and call this?

Reply:
Done, ended up refactoring out my strings -> map -> strings functions so I can use the delegating constructor here.

One hitch is that it makes the map and vector constructors ambiguous:

```
// Could be either
Subprocess({}, {})
```

So now you have to explicitly do:

```
Subprocess({}, gz::utils::EnvironmentMap{});
```

Not a huge deal, but a little less clean imo.